packages/keystone/KSv5v6/v5/prepareKeystone.js

```diff
@@ -115,7 +116,28 @@ function prepareKeystone ({ onConnect, extendKeystoneConfig, extendExpressApp, s
     // We need to register all schemas as they will appear in admin ui
     registerSchemas(keystone, schemas(), globalPreprocessors)
 
-    const authStrategyConfig = get(authStrategyOpts, 'config', {})
+    const defaultAuthStrategyConfigHooks = {
+        async afterAuth ({ item, token, success })   {
+            // NOTE: It's triggered only by default Keystone mutation, "authenticateUserWithPhoneAndPassword" will not work here
+            // Step 1. Skip if auth was not succeeded
+            if (!success || !token) {
+                return
+            }
+            // Step 2. Skip this for any non-service user, or users without type field
+            if (!item || !item.type || item.type !== 'service') {
+                return
+            }
+            // NOTE: auth token is just session token prefix (32 chars) + some prefix after dot.
+            // Example: 12345678901234567890123456789012.asdhaksdjhajskdhajskdhjkas
+            // Session token can be build like so "sess:{prefixBeforeDot}"
+            const sessToken = token.split('.')[0]
+            const sessKey = `sess:${sessToken}`
+            const redisClient = getRedisClient()
+            // NOTE: if key not found returns 0, else 1.
+            await redisClient.expire(sessKey, SERVICE_USER_SESSION_TTL_IN_SEC)
+        },
+    }
+    const authStrategyConfig = get(authStrategyOpts, 'config', { hooks: { afterAuth: [defaultAuthStrategyConfigHooks.afterAuth] } })
```

Comment:
Why to use this get hack? Why not to just compose all hooks together?

Reply:
this line (140) is my mistake. Config should be {} as before
In prepare let's join defined here hooks and hooks, provided from app
Getting many hooks of same type from app seems illogical, better let app handle their composing or merge before passing here